flake.nix

```diff
@@ -1,16 +1,16 @@
 {
   inputs = {
     nixpkgs.url = "github:NixOS/nixpkgs/release-21.11";
+    flake-utils.url = "github:numtide/flake-utils";
     nix-bundle-exe = {
       url = "github:3noch/nix-bundle-exe";
       flake = false;
     };
     gomod2nix = {
-      # https://github.com/tweag/gomod2nix/pull/38
-      url = "github:yihuang/gomod2nix/demonstrate-error";
+      url = "github:yihuang/gomod2nix/GOPROXY";
```

Comment:
https://github.com/tweag/gomod2nix/pull/43
wait for a small patch to be merged.